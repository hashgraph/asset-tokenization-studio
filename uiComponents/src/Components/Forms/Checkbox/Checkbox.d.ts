import type { checkboxAnatomy as ChakraCheckboxParts } from "@chakra-ui/anatomy";
import type { CheckboxProps as ChakraCheckboxProps } from "@chakra-ui/checkbox";
import React from "react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { SystemStyleObject as ChakraSystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
export declare const checkboxPartsList: Array<(typeof ChakraCheckboxParts.keys)[number] | "iconCustom">;
type Parts = typeof checkboxPartsList;
export interface CheckboxThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: Partial<CheckboxThemeStyle>;
}
export interface CheckboxProps extends Omit<ChakraCheckboxProps, "colorScheme" | "iconSize" | "isIndeterminate"> {
}
type CheckboxThemeStyle = Record<Parts[number], ChakraSystemStyleObject> & {
    iconCustom: React.FunctionComponent<{
        isChecked?: boolean;
    }>;
};
export declare const Checkbox: ComponentWithAs<"input", CheckboxProps>;
export {};
