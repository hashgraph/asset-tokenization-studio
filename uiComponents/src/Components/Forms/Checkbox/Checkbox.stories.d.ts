import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, any, import("./Checkbox").CheckboxProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const WithChildren: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, any, import("./Checkbox").CheckboxProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithOtherIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, any, import("./Checkbox").CheckboxProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const IsDisabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, any, import("./Checkbox").CheckboxProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const IsInvalid: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, any, import("./Checkbox").CheckboxProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const CheckboxController: () => JSX.Element;
