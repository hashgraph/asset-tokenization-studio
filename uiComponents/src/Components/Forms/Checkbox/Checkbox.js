var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Checkbox as ChakraCheckbox } from "@chakra-ui/checkbox";
import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Check } from "@phosphor-icons/react";
export var checkboxPartsList = ["container", "icon", "label", "control", "iconCustom"];
export var Checkbox = forwardRef(function (_a, ref) {
    var name = _a.name, isInvalid = _a.isInvalid, props = __rest(_a, ["name", "isInvalid"]);
    var formControl = useChakraFormControlContext() || {};
    var _b = useChakraMultiStyleConfig("Checkbox", __assign(__assign({}, props), { isInvalid: isInvalid })).iconCustom, ThemeIcon = _b === void 0 ? function () { return React.createElement(PhosphorIcon, { as: Check }); } : _b;
    var iconCustom = React.createElement(ThemeIcon, { isChecked: props.isChecked });
    return (React.createElement(ChakraCheckbox, __assign({ ref: ref, isInvalid: isInvalid !== null && isInvalid !== void 0 ? isInvalid : formControl.isInvalid, icon: iconCustom, name: name }, props)));
});
