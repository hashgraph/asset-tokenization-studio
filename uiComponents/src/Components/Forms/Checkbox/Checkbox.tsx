import type { checkboxAnatomy as ChakraCheckboxParts } from "@chakra-ui/anatomy";
import type { CheckboxProps as ChakraCheckboxProps } from "@chakra-ui/checkbox";
import { Checkbox as ChakraCheckbox } from "@chakra-ui/checkbox";
import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Check } from "@phosphor-icons/react";

export const checkboxPartsList: Array<
  (typeof ChakraCheckboxParts.keys)[number] | "iconCustom"
> = ["container", "icon", "label", "control", "iconCustom"];

type Parts = typeof checkboxPartsList;

export interface CheckboxThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle: Partial<CheckboxThemeStyle>;
}

export interface CheckboxProps
  extends Omit<
    ChakraCheckboxProps,
    "colorScheme" | "iconSize" | "isIndeterminate"
  > {}

type CheckboxThemeStyle = Record<Parts[number], ChakraSystemStyleObject> & {
  iconCustom: React.FunctionComponent<{ isChecked?: boolean }>;
};

export const Checkbox: ComponentWithAs<"input", CheckboxProps> = forwardRef<
  CheckboxProps,
  "input"
>(({ name, isInvalid, ...props }: CheckboxProps, ref) => {
  const formControl = useChakraFormControlContext() || {};
  const { iconCustom: ThemeIcon = () => <PhosphorIcon as={Check} /> } =
    useChakraMultiStyleConfig("Checkbox", {
      ...props,
      isInvalid,
    }) as CheckboxThemeStyle;
  const iconCustom = <ThemeIcon isChecked={props.isChecked} />;

  return (
    <ChakraCheckbox
      ref={ref}
      isInvalid={isInvalid ?? formControl.isInvalid}
      icon={iconCustom}
      name={name}
      {...props}
    />
  );
});
