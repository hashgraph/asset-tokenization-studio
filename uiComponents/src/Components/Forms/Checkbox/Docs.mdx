import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as CheckboxStories from "./Checkbox.stories";

<Meta of={CheckboxStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1874%3A15429&mode=design&t=KlerdyEMWSQWLAVX-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Checkbox.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Checkbox: ConfigCheckbox,
  },
};
```

```tsx dark
const ConfigCheckbox: CheckboxThemeConfiguration = {
  parts: checkboxPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"icon" | "label" | "container" | "control" | "iconCustom"` (`checkboxPartsList`)

## Props

<ArgTypes of={CheckboxStories} />

## Stories

<Stories title="" />
