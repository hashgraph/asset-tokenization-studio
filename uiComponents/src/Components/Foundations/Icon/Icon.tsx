import type { IconProps as ChakraIconProps } from "@chakra-ui/icon";
import type { IconProps as PhosphorIconProps } from "@phosphor-icons/react";
import { Icon as ChakraIcon } from "@chakra-ui/icon";
import { forwardRef } from "@chakra-ui/system";
import React from "react";
import type { CustomIconProps } from "./CustomIcon";
import { CustomIcon } from "./CustomIcon";
import type { ComponentWithAs } from "@chakra-ui/system";

export type IconButtonType = React.ForwardRefExoticComponent<
  PhosphorIconProps & React.RefAttributes<SVGSVGElement>
>;

export interface BaseIconProps
  extends Omit<ChakraIconProps, "name" | "css" | "weight"> {
  size?: string;
  variant?: string;
}

export type ConditionalIconProps =
  | { name?: false; as: any }
  | { name: CustomIconProps["name"]; as?: never };

export type IconProps = BaseIconProps & ConditionalIconProps;

export const Icon: ComponentWithAs<"svg", BaseIconProps & IconProps> =
  forwardRef<BaseIconProps & IconProps, "svg">(
    ({ name, ...props }: BaseIconProps & IconProps, ref) => {
      if (name) {
        return <CustomIcon ref={ref} name={name} {...props} />;
      }
      return <ChakraIcon ref={ref} {...props} />;
    }
  );
