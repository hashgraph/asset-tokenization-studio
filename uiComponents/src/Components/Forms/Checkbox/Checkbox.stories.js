var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Icon } from "@/Components/Foundations/Icon";
import { Checkbox } from "./Checkbox";
import { linkTo } from "@storybook/addon-links";
import { Button } from "@Components/Interaction/Button";
import { HouseLine } from "@phosphor-icons/react";
var meta = {
    title: "Design System/Forms/Checkbox",
    component: Checkbox,
    args: {
        variant: "square",
        size: "md"
    },
    argTypes: {
        children: { control: { type: "text" } },
        icon: { control: { type: null } },
        isInvalid: { control: { type: "boolean" } },
        isDisabled: { control: { type: "boolean" } },
        defaultChecked: { control: { type: "boolean" } }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1874%3A15429&t=GPCIOt6k4o0cCWIe-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return React.createElement(Checkbox, __assign({}, args)); };
export var WithChildren = Template.bind({});
WithChildren.args = {
    children: "I accept the Terms and Conditions & Privacy Policy"
};
export var WithOtherIcon = Template.bind({});
WithOtherIcon.args = {
    icon: React.createElement(Icon, { as: HouseLine })
};
export var IsDisabled = Template.bind({});
IsDisabled.args = {
    isDisabled: true
};
export var IsInvalid = Template.bind({});
IsInvalid.args = {
    isInvalid: true
};
export var CheckboxController = function () { return (React.createElement(Button, { onClick: linkTo("Design System/Forms/Controllers/CheckboxController") }, "Check out the CheckboxController component Stories")); };
