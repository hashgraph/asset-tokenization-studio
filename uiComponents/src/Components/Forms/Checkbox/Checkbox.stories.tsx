import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Icon } from "@/Components/Foundations/Icon";
import { Checkbox } from "./Checkbox";
import { linkTo } from "@storybook/addon-links";
import { Button } from "@Components/Interaction/Button";
import { HouseLine } from "@phosphor-icons/react";

const meta = {
  title: "Design System/Forms/Checkbox",
  component: Checkbox,
  args: {
    variant: "square",
    size: "md",
  },
  argTypes: {
    children: { control: { type: "text" } },
    icon: { control: { type: null } },
    isInvalid: { control: { type: "boolean" } },
    isDisabled: { control: { type: "boolean" } },
    defaultChecked: { control: { type: "boolean" } },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1874%3A15429&t=GPCIOt6k4o0cCWIe-4",
    },
    docs: {},
  },
} as Meta<typeof Checkbox>;
export default meta;

const Template: StoryFn<typeof Checkbox> = (args) => <Checkbox {...args} />;

export const WithChildren = Template.bind({});
WithChildren.args = {
  children: "I accept the Terms and Conditions & Privacy Policy",
};

export const WithOtherIcon = Template.bind({});
WithOtherIcon.args = {
  icon: <Icon as={HouseLine} />,
};

export const IsDisabled = Template.bind({});
IsDisabled.args = {
  isDisabled: true,
};

export const IsInvalid = Template.bind({});
IsInvalid.args = {
  isInvalid: true,
};

export const CheckboxController = () => (
  <Button
    onClick={linkTo("Design System/Forms/Controllers/CheckboxController")}
  >
    Check out the CheckboxController component Stories
  </Button>
);
