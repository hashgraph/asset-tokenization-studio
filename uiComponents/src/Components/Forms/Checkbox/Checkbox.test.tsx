import { render } from "@/test-utils";
import { Text as ChakraText } from "@chakra-ui/react";
import { Icon } from "@/Components/Foundations/Icon";
import type { RenderResult } from "@testing-library/react";
import { act, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import type { CheckboxProps } from "./Checkbox";
import { Checkbox } from "./Checkbox";
import { HouseLine } from "@phosphor-icons/react";

describe(`<Checkbox />`, () => {
  const testId = "test-checkbox";

  const textCheckbox = "checkbox item";
  const childrenProp = { children: <ChakraText>{textCheckbox}</ChakraText> };
  const sizeExtraSmallProp = { size: "xs" };
  const sizeSmallProp = { size: "sm" };
  const sizeMediumProp = { size: "md" };
  const sizeLargeProp = { size: "lg" };
  const iconProp = { icon: <Icon as={HouseLine} /> };
  const isDisabledProp = { isDisabled: true };
  const isInvalidProp = { isInvalid: true };
  const defaultCheckedProp = { defaultChecked: true };
  const squareVariantProp = { variant: "square" };
  const circleVariantProp = { variant: "circle" };

  const getInput = (component: RenderResult) =>
    component.getByTestId(testId).querySelector("input");

  const clickOnCheckbox = (component: RenderResult) => {
    act(() => {
      userEvent.click(getInput(component)!);
    });
  };

  const checkStatusRadio = (component: RenderResult, status: boolean) =>
    waitFor(() => {
      status
        ? expect(getInput(component)).toBeChecked()
        : expect(getInput(component)).not.toBeChecked();
    });

  const factoryComponent = (props?: CheckboxProps) =>
    render(<Checkbox name="test" data-testid={testId} {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly children", () => {
    const component = factoryComponent(childrenProp);

    expect(component.getByText(textCheckbox)).toBeVisible();

    expect(component.asFragment()).toMatchSnapshot("Using children");
  });

  test("shows new state when does click", async () => {
    const component = factoryComponent();

    clickOnCheckbox(component);

    await checkStatusRadio(component, true);
  });

  test("shows correctly when size is extra small", () => {
    const component = factoryComponent(sizeExtraSmallProp);

    expect(component.asFragment()).toMatchSnapshot("Using extra small size");
  });

  test("shows correctly when size is small", () => {
    const component = factoryComponent(sizeSmallProp);

    expect(component.asFragment()).toMatchSnapshot("Using small size");
  });

  test("shows correctly when size is medium", () => {
    const component = factoryComponent(sizeMediumProp);

    expect(component.asFragment()).toMatchSnapshot("Using medium size");
  });

  test("shows correctly when size is large", () => {
    const component = factoryComponent(sizeLargeProp);

    expect(component.asFragment()).toMatchSnapshot("Using large size");
  });

  test("shows correctly when is disabled", async () => {
    const component = factoryComponent(isDisabledProp);

    clickOnCheckbox(component);

    await checkStatusRadio(component, false);

    expect(component.asFragment()).toMatchSnapshot("Using as disabled");
  });

  test("shows correctly when is invalid", () => {
    const component = factoryComponent(isInvalidProp);

    expect(component.asFragment()).toMatchSnapshot("Using as invalid");
  });

  test("shows correctly when is checked by default", () => {
    const component = factoryComponent(defaultCheckedProp);

    expect(component.asFragment()).toMatchSnapshot("Using as default checked");
  });

  test("shows correctly when uses square variant", () => {
    const component = factoryComponent(squareVariantProp);

    expect(component.asFragment()).toMatchSnapshot("Using square variant");
  });

  test("shows correctly when uses circle variant", () => {
    const component = factoryComponent(circleVariantProp);

    expect(component.asFragment()).toMatchSnapshot("Using circle variant");
  });

  test("shows correctly when is checked and disabled", () => {
    const component = factoryComponent({
      ...defaultCheckedProp,
      ...isDisabledProp,
    });

    expect(component.asFragment()).toMatchSnapshot(
      "Using as default checked and disabled"
    );
  });

  test("shows correctly when uses other icon", () => {
    const component = factoryComponent(iconProp);

    expect(component.asFragment()).toMatchSnapshot("Using other icon");
  });
});
