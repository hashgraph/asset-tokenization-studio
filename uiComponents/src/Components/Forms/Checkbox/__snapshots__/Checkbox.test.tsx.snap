// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly children: Using children 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
    />
    <span
      class="chakra-checkbox__label css-tzpp0r"
    >
      <p
        class="chakra-text css-0"
      >
        checkbox item
      </p>
    </span>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when is checked and disabled: Using as default checked and disabled 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-checked=""
    data-disabled=""
    data-testid="test-checkbox"
  >
    <input
      aria-disabled="true"
      checked=""
      class="chakra-checkbox__input"
      disabled=""
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
      data-checked=""
      data-disabled=""
    >
      <svg
        class="chakra-icon css-1oeccli"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M229.66,77.66l-128,128a8,8,0,0,1-11.32,0l-56-56a8,8,0,0,1,11.32-11.32L96,188.69,218.34,66.34a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when is checked by default: Using as default checked 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-checked=""
    data-testid="test-checkbox"
  >
    <input
      checked=""
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
      data-checked=""
    >
      <svg
        class="chakra-icon css-1oeccli"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M229.66,77.66l-128,128a8,8,0,0,1-11.32,0l-56-56a8,8,0,0,1,11.32-11.32L96,188.69,218.34,66.34a8,8,0,0,1,11.32,11.32Z"
        />
      </svg>
    </span>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when is disabled: Using as disabled 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-disabled=""
    data-testid="test-checkbox"
  >
    <input
      aria-disabled="true"
      class="chakra-checkbox__input"
      disabled=""
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
      data-disabled=""
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when is invalid: Using as invalid 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-invalid=""
    data-testid="test-checkbox"
  >
    <input
      aria-invalid="true"
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
      data-invalid=""
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when size is extra small: Using extra small size 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-10vuwm"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when size is large: Using large size 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-1p4ng5h"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when size is medium: Using medium size 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when size is small: Using small size 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-1hm46cj"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when uses circle variant: Using circle variant 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-1suvs5d"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when uses other icon: Using other icon 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
    >
      <svg
        class="chakra-icon css-7kextp"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M240,208H224V115.55a16,16,0,0,0-5.17-11.78l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208H16a8,8,0,0,0,0,16H240a8,8,0,0,0,0-16ZM48,115.55l.11-.1L128,40l79.9,75.43.11.1V208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48ZM144,208H112V160h32Z"
        />
      </svg>
    </span>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Checkbox /> shows correctly when uses square variant: Using square variant 1`] = `
<DocumentFragment>
  <label
    class="chakra-checkbox css-1577qb8"
    data-testid="test-checkbox"
  >
    <input
      class="chakra-checkbox__input"
      name="test"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="checkbox"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-checkbox__control css-124d1"
    />
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
